benchmarks/BFS/NonDeterministicBFS/unit_tests/test_bfs.cc

```diff
@@ -0,0 +1,68 @@
+#include "benchmarks/BFS/NonDeterministicBFS/BFS.h"
+
+#include <unordered_set>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "ligra/graph.h"
+#include "ligra/graph_test_utils.h"
+#include "ligra/macros.h"
+
+using ::testing::AnyOf;
+using ::testing::ElementsAre;
+
+TEST(NondeterministicBFS, EdgelessGraph) {
+  constexpr uintE kNumVertices{3};
+  const std::unordered_set<UndirectedEdge> kEdges{};
+  auto graph{graph_test::MakeUnweightedSymmetricGraph(kNumVertices, kEdges)};
+
+  constexpr uintE source_vertex{1};
+  const pbbs::sequence<uintE> bfsResult{BFS(graph, source_vertex)};
+  EXPECT_THAT(bfsResult, ElementsAre(UINT_E_MAX, source_vertex, UINT_E_MAX));
+}
+
+TEST(NondeterministicBFS, BasicUsage) {
+  // Graph diagram:
+  //     0 - 1    2 - 3 - 4
+  //                    \ |
+  //                      5 -- 6
+  constexpr uintE kNumVertices{7};
+  const std::unordered_set<UndirectedEdge> kEdges{
+    {0, 1},
+    {2, 3},
+    {3, 4},
+    {3, 5},
+    {4, 5},
+    {5, 6},
+  };
+  auto graph{graph_test::MakeUnweightedSymmetricGraph(kNumVertices, kEdges)};
```

Comment:
Excellent. Thanks for the test. When we add more let's pull the graph building code out into some test_utils file to enable reuse. We can also write builders for a bunch of different graph classes (rMAT/barabasi-albert/line graphs, cliques, etc).